File: ai-event-gateway-admin-ui/scripts/verify-p9.5-skill-diff-impact-rbac.mjs
import fs from 'node:fs';
import path from 'node:path';

const root = process.cwd();
const checks = [
  ['lib/types/core.ts', 'CoreAgentSkillDiffResult'],
  ['lib/types/core.ts', 'CoreAgentSkillImpactAnalysisResult'],
  ['lib/types/core.ts', 'CoreAgentSkillApprovalPolicy'],
  ['lib/types/core.ts', 'operatorRoles'],
  ['lib/api/endpoints.ts', 'agentSkillVersionDiff'],
  ['lib/api/endpoints.ts', 'agentSkillVersionImpact'],
  ['lib/api/endpoints.ts', 'agentSkillApprovalPolicy'],
  ['lib/api/coreAdminApi.ts', 'getAgentSkillVersionDiff'],
  ['lib/api/coreAdminApi.ts', 'getAgentSkillVersionImpact'],
  ['lib/api/coreAdminApi.ts', 'updateAgentSkillApprovalPolicy'],
  ['components/skills/SkillRegistryConsole.tsx', 'P9.5 Approval RBAC'],
  ['components/skills/SkillRegistryConsole.tsx', 'Load Diff / Impact'],
  ['components/skills/SkillRegistryConsole.tsx', 'operatorRoles'],
];
const missing = [];
for (const [rel, token] of checks) {
  const full = path.join(root, rel);
  if (!fs.existsSync(full) || !fs.readFileSync(full, 'utf8').includes(token)) {
    missing.push(`${rel}: missing ${token}`);
  }
}
if (missing.length) {
  console.error(missing.join('\n'));
  process.exit(1);
}
console.log('P9.5 skill diff / impact / approval RBAC UI integration check passed.');

File: ai-event-gateway-admin-ui/scripts/verify/verify-p9.5-skill-diff-impact-rbac.py
#!/usr/bin/env python3
from pathlib import Path
import sys
import xml.etree.ElementTree as ET

ROOT = Path(__file__).resolve().parents[2]
checks = [
    ('ai-event-gateway-core-agent-control/src/main/java/com/opensocket/aievent/core/agent/skill/AgentSkillDiffEntry.java', 'class AgentSkillDiffEntry'),
    ('ai-event-gateway-core-agent-control/src/main/java/com/opensocket/aievent/core/agent/skill/AgentSkillDiffResult.java', 'breakingFields'),
    ('ai-event-gateway-core-agent-control/src/main/java/com/opensocket/aievent/core/agent/skill/AgentSkillImpactAnalysisResult.java', 'impactedAgents'),
    ('ai-event-gateway-core-agent-control/src/main/java/com/opensocket/aievent/core/agent/skill/AgentSkillApprovalPolicy.java', 'separationOfDuties'),
    ('ai-event-gateway-core-agent-control/src/main/java/com/opensocket/aievent/core/agent/skill/AgentSkillWorkflowCommand.java', 'operatorRoles'),
    ('ai-event-gateway-core-agent-control/src/main/java/com/opensocket/aievent/core/agent/skill/AgentSkillRegistryService.java', 'diffVersion'),
    ('ai-event-gateway-core-agent-control/src/main/java/com/opensocket/aievent/core/agent/skill/AgentSkillRegistryService.java', 'analyzeImpact'),
    ('ai-event-gateway-core-agent-control/src/main/java/com/opensocket/aievent/core/agent/skill/AgentSkillRegistryService.java', 'enforceWorkflowRole'),
    ('ai-event-gateway-core-app/src/main/java/com/opensocket/aievent/core/api/AgentSkillRegistryController.java', '/admin/agent-skills/{skillCode}/versions/{version}/diff'),
    ('ai-event-gateway-core-app/src/main/java/com/opensocket/aievent/core/api/AgentSkillRegistryController.java', '/admin/agent-skills/{skillCode}/approval-policy'),
    ('ai-event-gateway-database-platform/src/main/java/com/opensocket/aievent/database/persistence/agent/skill/po/AgentSkillApprovalPolicyPo.java', 'class AgentSkillApprovalPolicyPo'),
    ('ai-event-gateway-database-platform/src/main/resources/db/migration/V31__agent_skill_approval_policy.sql', 'agent_skill_approval_policies'),
    ('ai-event-gateway-database-platform/src/main/resources/mybatis/postgresql/agent/AgentSkillRegistryDao.xml', 'SkillApprovalPolicyMap'),
]
missing = []
for rel, token in checks:
    path = ROOT / rel
    if not path.exists() or token not in path.read_text():
        missing.append(f'{rel}: missing {token}')
xml_path = ROOT / 'ai-event-gateway-database-platform/src/main/resources/mybatis/postgresql/agent/AgentSkillRegistryDao.xml'
try:
    ET.parse(xml_path)
except Exception as exc:
    missing.append(f'{xml_path}: XML parse failed: {exc}')
if missing:
    print('\n'.join(missing), file=sys.stderr)
    sys.exit(1)
print('P9.5 skill diff / impact analysis / approval RBAC integration check passed.')
